conductor_api/src/conductor/passphrase_manager.rs

```diff
@@ -125,3 +129,86 @@ impl PassphraseService for PassphraseServiceMock {
         Ok(SecBuf::with_insecure_from_string(self.passphrase.clone()))
     }
 }
+
+#[cfg(unix)]
+pub struct PassphraseServiceUnixSocket {
+    path: String,
+    stream: Arc<Mutex<Option<std::io::Result<BufReader<UnixStream>>>>>,
+}
+
+#[cfg(unix)]
+impl PassphraseServiceUnixSocket {
+    pub fn new(path: String) -> Self {
+        let stream = Arc::new(Mutex::new(None));
+        let stream_clone = stream.clone();
+        let listener = UnixListener::bind(path.clone())
+            .expect("Could not create unix socket for passphrase service");
+        log_info!("Start accepting passphrase IPC connections on socket...");
+        thread::spawn(move || {
+            let accept_result = listener.accept();
+            {
+                *(stream_clone.lock().unwrap()) =
+                    Some(accept_result.map(|(stream, _)| BufReader::new(stream)));
+            }
+            log_info!("Passphrase provider connected through unix socket");
+        });
+        PassphraseServiceUnixSocket { path, stream }
+    }
+}
+
+#[cfg(unix)]
+impl Drop for PassphraseServiceUnixSocket {
+    fn drop(&mut self) {
+        std::fs::remove_file(self.path.clone()).unwrap();
+    }
+}
+
+#[cfg(unix)]
+impl PassphraseService for PassphraseServiceUnixSocket {
+    fn request_passphrase(&self) -> Result<SecBuf, HolochainError> {
+        log_debug!("Passphrase needed. Using unix socket passphrase service...");
+        while self.stream.lock().unwrap().is_none() {
```

Comment:
I think we probably should timeout here at some point no?

Reply:
The only reasonable thing to do then would be to panic.. why not keep waiting?